### PR TITLE
parser: remove old parser and move ast/parser to /parser.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

File: parser/parser_test.go [hunk 1]
```diff
@@ -18,9 +5,7 @@ import (
 	"testing"
 
 	. "github.com/pingcap/check"
-	"github.com/pingcap/tidb/expression"
-	"github.com/pingcap/tidb/expression/subquery"
-	"github.com/pingcap/tidb/stmt/stmts"
+	"github.com/pingcap/tidb/ast"
```
Comment (qiuyesuifeng) [inline parser/parser_test.go]: Add license header.

File: parser/parser_test.go [hunk 2]
```diff
@@ -29,24 +14,9 @@ func TestT(t *testing.T) {
...
 
-type testParserSuite struct {
-}
-
-func (s *testParserSuite) TestOriginText(c *C) {
-	src := `SELECT stuff.id 
-		FROM stuff 
-		WHERE stuff.value >= ALL (SELECT stuff.value 
-		FROM stuff)`
+var _ = Suite(&testParserSuite{})
```
Comment (qiuyesuifeng) [inline parser/parser_test.go]: Has been defined in line 15.

File: parser/parser_test.go [hunk 3]
```diff
@@ -58,7 +28,7 @@ func (s *testParserSuite) TestSimple(c *C) {
 		"start", "global", "tables", "text", "time", "timestamp", "transaction", "truncate", "unknown",
 		"value", "warnings", "year", "now", "substring", "mode", "any", "some", "user", "identified",
 		"collation", "comment", "avg_row_length", "checksum", "compression", "connection", "key_block_size",
-		"max_rows", "min_rows", "national", "row", "quarter", "escape", "grants",
+		"max_rows", "min_rows", "national", "row", "quarter", "escape",
```
Comment (qiuyesuifeng) [inline parser/parser_test.go]: Seems `grants` be removed?

File: parser/scanner.l [hunk 4]
```diff
@@ -1047,7 +1056,7 @@ func (l *lexer) str(lval *yySymType, pref string) int {
 		pref = "\""
 	} 
 	v := stringutil.RemoveUselessBackslash(pref+s)
-	v, err := strconv.Unquote(v)
+        v, err := strconv.Unquote(v)
```
Comment (qiuyesuifeng) [inline parser/scanner.l]: alignment

File: Makefile [hunk 5]
```diff
@@ -13,7 +13,7 @@ TARGET = ""
 
 .PHONY: godep deps all build install parser clean todo test gotest interpreter server
 
-all: godep parser ast-parser build test check
+all: godep parser build test check
```
Comment (qiuyesuifeng) [inline Makefile]: Should we remove `ast-parser` from make.cmd?

**Comment by ngaut** [comment]
LGTM

**Pull request merged**